libs/kotlinlib/src/mill/kotlinlib/ksp/Ksp2Module.scala

```diff
@@ -0,0 +1,156 @@
+package mill.kotlinlib.ksp
+
+import coursier.core.VariantSelector.VariantMatcher
+import coursier.params.ResolutionParams
+import mill.*
+import mill.api.Result
+import mill.api.{PathRef, Task}
+import mill.kotlinlib.{Dep, DepSyntax}
+import mill.util.Jvm
+
+import java.io.File
+
+/**
+ * Sets up the kotlin compiler for using KSP 2.x (Kotlin Symbol Processing)
+ * This module is based on KSP 2.x which supports language version 2.0 and later.
+ * For KSP 1.x, use [[KspModule]] instead.
+ *
+ *  Documentation: https://github.com/google/ksp/blob/main/docs/ksp2cmdline.md
+ */
+@mill.api.experimental
+trait Ksp2Module extends KspBaseModule { outer =>
+
+  /**
+   * The JVM target version for the KSP compilation step.
+   * Default is derived from `java.specification.version`.
+   */
+  def kspJvmTarget: T[String] = Task {
+    System.getProperty("java.specification.version")
+  }
+
+  /**
+   * The kotlin language version used for the KSP compilation stage.
+   * [[kspApiVersion]] must be less than or equal to this version.
+   * @return
+   */
+  def kspLanguageVersion: T[String] = kotlinLanguageVersion()
+
+  /**
+   * The kotlin api version used for the KSP compilation stage.
+   * @return
+   */
+  def kspApiVersion: T[String] = kotlinApiVersion()
+
+  /**
+   * The jars needed to run KSP 2 via `com.google.devtools.ksp.cmdline.KSPJvmMain` .
+   *
+   * The versions are computed from [[kotlinVersion]]-[[kspVersion]]
+   *
+   * For finding the right versions, also see [[https://github.com/google/ksp/releases]]
+   * For more info go to [[https://github.com/google/ksp/blob/main/docs/ksp2cmdline.md]]
+   */
+  def kspDeps: T[Seq[Dep]] = Task {
```

Comment:
So these deps are rather some "internal" tooling classpath but nothing the user needs to configure?

Reply:
yes, should we have them private?

Reply:
Let keep them public, so users can use locally modified tools. Something, that Mill is really predestined for.

Reply:
ok, will change to `kspToolsDeps` and `kspToolsClasspath`